internal/policies/apparmor/apparmor.go

```diff
@@ -0,0 +1,262 @@
+	// If the old apparmor directory is not present, it means we succeeded in
+	// applying the new policy
+	if _, err := os.Stat(oldApparmorPath); err != nil && os.IsNotExist(err) {
+		return
+	}
+
+	// Otherwise, we need to restore the old apparmor directory
+	if err := os.RemoveAll(apparmorPath); err != nil {
+		log.Warningf(ctx, "can't remove new apparmor directory: %v", err)
+	}
+
+	if err := os.Rename(oldApparmorPath, apparmorPath); err != nil {
+		log.Warningf(ctx, "can't restore previous apparmor directory: %v", err)
+	}
+}
+
+// unloadRules unloads the apparmor rules in the given directory and removes the directory.
+// No action is taken if the directory doesn't exist.
+func unloadRules(ctx context.Context, apparmorParserCmd string, apparmorPath string) (err error) {
+	defer decorate.OnError(&err, i18n.G("can't unload apparmor rules"))
+
+	// Nothing to do if the directory doesn't exist
+	if _, err := os.Stat(apparmorPath); err != nil && os.IsNotExist(err) {
+		return nil
+	}
+
+	// Walk the directory and get all the files to unload
+	var filesToUnload []string
+	if err := filepath.WalkDir(apparmorPath, func(path string, d os.DirEntry, err error) error {
+		if !d.Type().IsRegular() {
+			return nil
+		}
+		filesToUnload = append(filesToUnload, path)
+		return nil
+	}); err != nil {
+		return err
+	}
+
+	args := []string{"-R"}
+	args = append(args, filesToUnload...)
+
+	// #nosec G204 - We are in control of the arguments
+	cmd := exec.CommandContext(ctx, apparmorParserCmd, args...)
+	if out, err := cmd.CombinedOutput(); err != nil {
+		return fmt.Errorf(i18n.G("failed to unload apparmor rules: %w\n%s"), err, string(out))
+	}
+
+	// Unloading succeeded, remove apparmor policy dir
+	if err := os.RemoveAll(apparmorPath); err != nil {
+		log.Warningf(ctx, "can't remove apparmor directory: %v", err)
+	}
+	return nil
+}
+
+// filesFromEntry returns the list of files configured in the given policy entry.
+// It returns an error if the file does not exist or is a directory.
+func filesFromEntry(e entry.Entry, apparmorPath string) ([]string, error) {
+	var filesToLoad []string
+	for _, script := range strings.Split(e.Value, "\n") {
+		script = strings.TrimSpace(script)
+		if script == "" {
+			continue
+		}
+
+		scriptFilePath := filepath.Join(apparmorPath, script)
+		info, err := os.Stat(scriptFilePath)
+		if errors.Is(err, os.ErrNotExist) {
+			return nil, fmt.Errorf(i18n.G("script %q doesn't exist in SYSVOL apparmor/ subdirectory"), script)
+		}
+		if info.IsDir() {
+			return nil, fmt.Errorf(i18n.G("script %q is a directory and not a file to execute"), script)
+		}
+
+		filesToLoad = append(filesToLoad, filepath.Clean(scriptFilePath))
+	}
+	return filesToLoad, nil
+}
+
+// removeUnusedAssets removes all files/directories in the given directory that
+// are not in the given list of files.
+func removeUnusedAssets(apparmorPath string, filesToKeep []string) error {
```

Comment:
I would decorate the error here and just return directly the err in each branch, as the error should be descriptive enough.